### PR TITLE
SolarHelmets

### PR DESCRIPTION
**Pull request by J3fftw1**
## Short Description
Added some upgraded versions from the solar helmets
recipe still needs changing

## Additions/Changes/Removals
Adavanced Solar Helmet
Carbonado Solar Helmet
Energized Solar Helmet
Super Advanced Solar Helmet
Hybrid Solar Helmet
Ultimate Solar Helmet
Iron Plate
Diamond Plate
Thorium Plate

## Checklist

- [x] I have fully tested the proposed changes and promise that they will not break everything into chaos.
- [x] I have also tested the proposed changes in combination with base Slimefun and made sure nothing breaks/unexpected happens.
- [x] I followed the existing code standards and didn't mess up the formatting.
- [ ] I did my best to add documentation to any public classes or methods I added which may not be obvious to maintainers.
- [x] I have added `Nonnull` and `Nullable` annotations to my methods to indicate their behaviour for null values

### CODE REVIEW

File: src/main/java/dev/j3fftw/litexpansion/armor/SolarHelmets.java [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import javax.annotation.Nonnull;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.inventory.ItemStack;
+
+public class SolarHelmets extends SolarHelmet {
```
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/armor/SolarHelmets.java]: Conventions :NotLikeThis: No plural

```suggestion
public class AdvancedSolarHelmet extends SolarHelmet {
```

File: src/main/java/dev/j3fftw/litexpansion/armor/SolarHelmets.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        super(Items.LITEXPANSION, type.getItem(), RecipeType.ENHANCED_CRAFTING_TABLE, type.getRecipe(),
+            type.getGenerationRate()
+        );
+        this.type = type;
+        Items.ADVANCED_SOLAR_HELMET.addEnchantment(Enchantment.DURABILITY, 1);
+        Items.CARBONADO_SOLAR_HELMET.addEnchantment(Enchantment.DURABILITY, 2);
+        Items.ENERGIZED_SOLAR_HELMET.addEnchantment(Enchantment.DURABILITY, 3);
+        Items.ADVANCEDLX_SOLAR_HELMET.addUnsafeEnchantment(Enchantment.DURABILITY, 4);
+        Items.HYBRID_SOLAR_HELMET.addUnsafeEnchantment(Enchantment.DURABILITY, 5);
+        Items.ULTIMATE_SOLAR_HELMET.addUnsafeEnchantment(Enchantment.DURABILITY, 6);
```
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/armor/SolarHelmets.java]: This is redundant as it runs everytime a new solar helmet is initialized.
You are adding these enchantments over and over again.
This should probably happen once only in the Items class as a static call

File: src/main/java/dev/j3fftw/litexpansion/armor/SolarHelmets.java [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+import me.mrCookieSlime.Slimefun.Lists.RecipeType;
+import me.mrCookieSlime.Slimefun.api.SlimefunItemStack;
+import org.bukkit.enchantments.Enchantment;
+import org.bukkit.inventory.ItemStack;
+
+public class SolarHelmets extends SolarHelmet {
+
+    private final Type type;
+
+    public SolarHelmets(Type type) {
```
Comment (TheBusyBiscuit) [inline src/main/java/dev/j3fftw/litexpansion/armor/SolarHelmets.java]: Also could use a Nonnull annotation

**Review by TheBusyBiscuit — CHANGES_REQUESTED** [review]

File: src/main/java/dev/j3fftw/litexpansion/armor/SolarHelmets.java (hunk 1 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/armor/SolarHelmets.java]: is this an rule or something havent heard that before

File: src/main/java/dev/j3fftw/litexpansion/armor/SolarHelmets.java (hunk 2 above)
Comment (J3fftw1) [reply src/main/java/dev/j3fftw/litexpansion/armor/SolarHelmets.java]: How would i do it then?

**Pull request merged**